src/nav/nav.spec.ts

```diff
@@ -1094,6 +1150,253 @@ describe('nav', () => {
+      links[2].click();
+      fixture.detectChanges();
+
+      // 6. links are updated synchronously
+      expectLinks(fixture, [false, false, true]);
+
+      // 7. removing 2nd, adding 3rd content, 1st still active
+      expectContents(fixture, ['content 1', 'content 3'], 0);
+      [first, second] = getContents(fixture);
+      expectContentState(first, ['fade'], ['show'], '1');
+      expectContentState(second, ['fade'], ['show'], '0');
+    });
+
+    it(`should reverse fade in if switched to a new pane after fading in has started already`, (done) => {
+      const fixture = TestBed.createComponent(TestAnimationComponent);
+      const {onItemHiddenSpy, onItemShownSpy, onNavHiddenSpy, onNavShownSpy} = fixture.componentInstance;
+      fixture.componentInstance.reduceMotion = false;
+      fixture.detectChanges();
+
+      // TEST: 1 is active, switching 1 -> 2, switching 2 -> 3 while fading in 2, catching nav (shown)
+      let first: HTMLElement;
+      let second: HTMLElement;
+
+      // 1. checking links have good initial values (1 is active)
+      const links = getLinks(fixture);
+      expectLinks(fixture, [true, false, false]);
+      expectContents(fixture, ['content 1']);
+      expectContentState(getContent(fixture), ['fade', 'show'], [], '1');
+
+      onNavHiddenSpy.and.callFake(hiddenId => {
+        // 5. (hidden) is fired 2 times, for 1 -> 2 them for 2 -> 3
+        // we care only about the 1 -> 2
+        if (hiddenId === 1) {
+          fixture.detectChanges();
+          expect(onItemHiddenSpy).toHaveBeenCalledWith(1);
+          expectContents(fixture, ['content 2']);
+          expectContentState(getContent(fixture), ['fade', 'show'], [], '0');
+
+          // 6. switching 2 -> 3
+          links[2].click();
+          fixture.detectChanges();
+
+          // 7. links are updated synchronously
+          expectLinks(fixture, [false, false, true]);
+
+          // 8. adding 3rd content, 2nd still active (only starting fading in)
+          expectContents(fixture, ['content 2', 'content 3'], 0);
+          [first, second] = getContents(fixture);
+          expectContentState(first, ['fade'], ['show'], '0');
+          expectContentState(second, ['fade'], ['show'], '0');
+        }
+      });
+
+      onNavShownSpy.and.callFake(shownId => {
+        // 9. (shown) is fired 2 times, for 1 -> 2, then for 2 -> 3
+        // we care only about 2 -> 3
+        if (shownId === 3) {
+          fixture.detectChanges();
+          expect(shownId).toBe(3);
+          expect(onNavHiddenSpy).toHaveBeenCalledWith(1);
+          expect(onNavHiddenSpy).toHaveBeenCalledWith(2);
+          expect(onItemHiddenSpy).toHaveBeenCalledWith(1);
+          expect(onItemHiddenSpy).toHaveBeenCalledWith(2);
+          expect(onItemShownSpy).toHaveBeenCalledWith(3);
+          expectContents(fixture, ['content 3']);
+          expectContentState(getContent(fixture), ['fade', 'show'], [], '1');
+          done();
```

Comment:
Following my previous comment, maybe you should add (in each test) checks like:

```
expect(onNavHiddenSpy).toHaveBeenCalledTimes(x);
expect(onNavShownSpy).toHaveBeenCalledTimes(x);
```
just before the `done()`, to be sure that there is no extra call/work done before the test end.

What do you think ?

Reply:
you mean just to make sure that is called more than two times?

this kind of imlies that is is called two times:

```ts
        expect(onItemHiddenSpy).toHaveBeenCalledWith(1);
        expect(onItemHiddenSpy).toHaveBeenCalledWith(2);
```